src/main/java/org/libertybikes/ai/service/RegistrationBean.java

```diff
@@ -0,0 +1,81 @@
+package org.libertybikes.ai.service;
+
+import java.net.URI;
+import java.util.Objects;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.json.bind.Jsonb;
+import javax.json.bind.JsonbBuilder;
+import javax.json.bind.annotation.JsonbProperty;
+import javax.websocket.ContainerProvider;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.sse.InboundSseEvent;
+import javax.ws.rs.sse.SseEventSource;
+
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+import org.libertybikes.ai.restclient.GameServiceClient;
+
+/**
+ * This bean handles getting the bot into its first round and then requeueing
+ * the bot to the next round after the current round has completed
+ */
+@ApplicationScoped
+public class RegistrationBean {
+
+    private static final Jsonb jsonb = JsonbBuilder.create();
+
+    @Inject
+    AIConfiguration config;
+
+    @Inject
+    GameServiceClient gameService;
+
+    public void joinRound() {
+        System.out.println("Attempting to register with game service...");
+
+        String partyId = (String) gameService.describe().get("partyId");
+        System.out.println("Found party id: " + partyId);
+
+        // MP Rest Client doesn't yet support JAX-RS server-sent-events, so we nee to do this manually
+        // enhancement tracked at: https://github.com/eclipse/microprofile-rest-client/issues/11
+        String targetStr = config.getGameServiceHttp() + "/party/" + partyId + "/queue?playerId=" + config.getPlayerId();
+        System.out.println("Establishing SSE target at: " + targetStr);
+        Client client = ClientBuilder.newClient();
+        WebTarget target = client.target(targetStr);
+        SseEventSource source = SseEventSource.target(target).build();
+        source.register(inboundEvent -> processInboundEvent(inboundEvent, source),
+                        errEvent -> errEvent.printStackTrace(),
+                        () -> System.out.println("Closing SSE source for " + targetStr));
+        source.open();
+        // intentionally "leak" the source, it will be closed in processInboundEvent() when the proper event is received
```

Comment:
lets remove these lines and make it part of section 2 so that they get exposure to JAX-RS SSE